src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,69 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.spark.config.EncoderConfig;
+import com.spikes2212.command.genericsubsystem.smartmotorcontrollersubsystem.SmartMotorControllerGenericSubsystem;
+import com.spikes2212.util.smartmotorcontrollers.SparkWrapper;
+import edu.wpi.first.wpilibj.DigitalInput;
+
+import java.util.function.Supplier;
+
+public class Elevator extends SmartMotorControllerGenericSubsystem {
+
+    public enum ElevatorLevels {
+        processor(-1), feeder(-1), L1(-1), L2(-1), L3(-1), L4(-1);
+
+        public final double height;
+
+        ElevatorLevels(double height) {
+            this.height = height;
+        }
+    }
+
+    private static final double GEAR_RATIO = (14 / 50.0) * (16 / 50.0);
+    private static final double SPINS_TO_HEIGHT = 2 / (GEAR_RATIO * 41.2 * Math.PI);
+
+    private final SparkWrapper master;
+
+    private final DigitalInput minLimit;
+    private final DigitalInput maxLimit;
+
+    public Elevator(String namespaceName, SparkWrapper master, SparkWrapper slave, DigitalInput minLimit,
+                    DigitalInput maxLimit) {
+        super(namespaceName, master, slave);
+        this.master = master;
+        this.minLimit = minLimit;
+        this.maxLimit = maxLimit;
+        slave.setInverted(true);
+
+        EncoderConfig encoderConfig = new EncoderConfig();
+        encoderConfig.positionConversionFactor(SPINS_TO_HEIGHT);
```

Comment:
instead of this, call the `setPositionConversionFactor()` function in `SparkWrapper`